sdk/go/common/apitype/migrate/resource.go

```diff
@@ -51,6 +53,8 @@ func UpToResourceV2(v1 apitype.ResourceV1) apitype.ResourceV2 {
 
 // UpToResourceV3 migrates a resource from ResourceV2 to ResourceV3.
 func UpToResourceV3(v2 apitype.ResourceV2) apitype.ResourceV3 {
+	now := time.Now()
```

Comment:
This is going to make me sound like I'm wearing a tinfoil hat, but I'd suggest _not_ reusing `now`'s pointer in the struct below (and above).

It sounds crazy, but we've had a number of nasty data races where one thread would be iterating over the state, while another thread would be modifying it, resulting in a concurrent map mutation error.
```suggestion
	var now = time.Now()
	var created, updated = now, now // by-value copy of the time.Time struct.
```

Reply:
Happy to do whatever. I removed this instance based on @Frassle's comment about defaults but kept it in a later case.

Would the race detector catch this?